pkg/config/validator.go

```diff
@@ -421,9 +550,83 @@ func (p *validator) validate(cfg map[string]string) (rt *Validated, ce *adapter.
 			return rt, ce.Appendf("serviceConfig", "failed validation").Extend(re)
 		}
 	}
+
+	for _, kk := range keymap[constructors] {
+		ck := parseRulesKey(kk)
+		if ck == nil {
+			continue
+		}
+		if re := p.validateConstructorConfigs(cfg[kk]); re != nil {
+			return rt, ce.Appendf("serviceConfig", "failed validation").Extend(re)
```

Comment:
I know you're following the pattern of the rest of this func, but Is there a reason we have to return here and elsewhere in this func? It feels like we could keep going and wind up aggregating more errors in a single pass. Nothing is a worse debugging experience than running, getting a set of errors, then running again to get more new errors. We should try to report as many errors as possible in a single pass.